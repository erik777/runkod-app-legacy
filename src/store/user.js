export const USER_LOGIN = '@user/LOGIN';
export const USER_LOGOUT = '@user/USER_LOGOUT';

const initialState = '';

/* Reducer */

export default (state = initialState, action) => {
  switch (action.type) {
    case USER_LOGIN:
      return action.payload;
    case USER_LOGOUT:
      return initialState;
    default:
      return state;
  }
}

/* Actions */
export const login = (username) => {
  return async (dispatch) => {
    dispatch(loginAct(username));
  }
};

export const logout = () => {
  return async (dispatch) => {
    dispatch(logOutAct());
  }
};


/* Action creators */
export const loginAct = (username) => ({
  type: USER_LOGIN,
  payload: username
});


export const logOutAct = () => ({
  type: USER_LOGOUT
});