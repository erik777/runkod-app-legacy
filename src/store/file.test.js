import reducer, {selectAct} from './file'
import {logOutAct} from './user';

let state = undefined;

it('1- default state', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- select', () => {
  const act = selectAct({_id: 'project1', name: 'x.png', status: 1});
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('3- user logout', () => {
  const act = logOutAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
