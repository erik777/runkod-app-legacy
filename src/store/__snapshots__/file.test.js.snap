// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `null`;

exports[`2- select 1`] = `
Object {
  "_id": "project1",
  "name": "x.png",
  "status": 1,
}
`;

exports[`3- user logout 1`] = `null`;
